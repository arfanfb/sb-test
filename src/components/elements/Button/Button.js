import React from 'react';
import PropTypes, { object } from 'prop-types';
import { Link } from 'react-router-dom';
import Typography from '../Typography';
import styles from './Button.module.scss';

export default function Button(props) {
  const { children, 'class-name': className, disabled, height,
    icon, onClick, to, type, variant, width } = props;
  const customClass = [styles['button'], styles[variant], className].filter(Boolean).join(' ');
  const content = typeof children === object ? <Typography>{children}</Typography> : children;

  return (to ?
    (<Link className={customClass} disabled={disabled} href={to} style={{ width: width }}>
      <a><span>{icon} {content}</span></a></Link>) :
    (<button className={customClass} disabled={disabled}
      onClick={onClick} style={{ height: height, width: width }} type={type}>
      {icon} {content}</button>)
  );
}

Button.defaultProps = {
  'class-name': '',
  children: null,
  disabled: false,
  height: 48,
  icon: null,
  onClick: null,
  to: undefined,
  type: 'button',
  variant: 'primary',
  width: 182,
};

Button.propTypes = {
  children: PropTypes.node,
  'class-name': PropTypes.string,
  disabled: PropTypes.bool,
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  icon: PropTypes.node,
  onClick: PropTypes.func,
  to: PropTypes.string,
  type: PropTypes.string,
  variant: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};
