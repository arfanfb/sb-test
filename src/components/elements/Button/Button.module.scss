@import '../../../styles/colors.scss';

.primary[disabled]{
  background-color: map-get($primary, light);
  border: 0.0625rem solid map-get($primary, light);
  pointer-events: none;
}
.secondary[disabled]{
  color: map-get($primary, light);
  pointer-events: none;
}
.outline[disabled]{
  color: map-get($primary, light);
  border: 0.0625rem solid map-get($primary, light);
  pointer-events: none;
}
.button{
  text-align: center;
  align-items: center;
  justify-content: center;
  outline: none;
  display: flex;
}
.primary, .primary:focus{
  background-color: map-get($primary , base);
  border: 0.0625rem solid map-get($primary , base);
  color: $white-color;
}
.primary, .primary:focus,.secondary,.secondary:focus {
  width: 182px;
  height: 42px;
  border-radius: 100px;
  font-weight: 700;
  font-size: 14px;
}
.secondary,.secondary:focus{
  background-color: $white-color;
  border: 0.0625rem solid $white-color;
  color: map-get($primary, base);
}

.primary:not(:disabled):hover {
  background-color: map-get($primary, dark);
  border-color:map-get($primary, dark);
  cursor: pointer;
}
.secondary:not(:disabled):hover{
  background-color: $white-color;
  border-color: $white-color;
  color: map-get($primary , dark);
  cursor: pointer;
}

.outline,.outline:focus{
  background-color: $white-color;
  border: 0.0625rem solid map-get($primary, base);
  width: 182px;
  height: 42px;
  border-radius: 100px;
  font-size: 14px;
  font-weight: 700;
  color: map-get($primary, base);
}

.outline:not(:disabled):hover {
  border-color:map-get($primary, dark);
  color: map-get($primary, dark);
  cursor: pointer;
}
