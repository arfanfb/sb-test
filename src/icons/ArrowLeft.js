import React from "react";
import PropTypes from "prop-types";

export default function ArrowLeft(props) {
  const { className: classNameProps, fill, height, onClick, width } = props;
  return (
    <svg
      className={classNameProps} height={height} onClick={onClick}
      viewBox={`0 0 ${height} ${width}`} xmlns="http://www.w3.org/2000/svg" width={width}
    >
      <path clipRule="evenodd" 
        d={`M12.6666 7.33344H4.75663L7.17863 4.42677C7.41463 4.1441 7.37596 3.72344 7.09329 3.4881C6.80996
          3.2521 6.38996 3.29077 6.15463 3.57344L2.82129 7.57344C2.79529 7.60477 2.78196 7.64144 2.76263
          7.6761C2.74663 7.7041 2.72729 7.7281 2.71529 7.75877C2.68529 7.83544 2.66729 7.9161 2.66729
          7.99744C2.66729 7.9981 2.66663 7.99944 2.66663 8.0001C2.66663 8.00077 2.66729 8.0021 2.66729
          8.00277C2.66729 8.0841 2.68529 8.16477 2.71529 8.24144C2.72729 8.2721 2.74663 8.2961 2.76263
          8.3241C2.78196 8.35877 2.79529 8.39544 2.82129 8.42677L6.15463 12.4268C6.28663 12.5848 6.47596
          12.6668 6.66663 12.6668C6.81729 12.6668 6.96863 12.6161 7.09329 12.5121C7.37596 12.2768 7.41463
          11.8561 7.17863 11.5734L4.75663 8.66677H12.6666C13.0346 8.66677 13.3333 8.3681 13.3333
          8.0001C13.3333 7.6321 13.0346 7.33344 12.6666 7.33344Z`} fill={fill}/>
    </svg>
  );
}

ArrowLeft.defaultProps = {
  fill: '#FFFFFF',
  height: '16',
  onClick: () => {},
  width: '16'
};
ArrowLeft.propTypes = {
  fill: PropTypes.string,
  height: PropTypes.string,
  onClick: PropTypes.func,
  width: PropTypes.string
};

