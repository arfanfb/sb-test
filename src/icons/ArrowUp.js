import React from 'react';
import PropTypes from 'prop-types';

export default function ArrowUp(props) {
  const { fill } = props;
  return (
    <svg fill="none" height="12" viewBox="0 0 22 12" width="22" xmlns="http://www.w3.org/2000/svg" {...props}>
      <path d="M20.1721 11.0796C20.2756 11.1774 20.3972 11.2538 20.5302 11.3045C20.6631 11.3552 20.8048 11.3792 20.947 11.3752C21.0893
      11.3712 21.2293 11.3392 21.3592 11.281C21.4891 11.2229 21.6062 11.1397 21.704 11.0363C21.8017 10.9329 21.8781 10.8112 21.9288
      10.6782C21.9796 10.5453 22.0036 10.4036 21.9996 10.2614C21.9955 10.1192 21.9635 9.97909 21.9054 9.84922C21.8472 9.71934 21.7641
      9.60219 21.6606 9.50445L11.9106 0.296121C11.7095 0.105956 11.4432 -4.61452e-07 11.1664 -4.73552e-07C10.8896 -4.85651e-07 10.6233
      0.105956 10.4221 0.296121L0.67106 9.50445C0.565368 9.60154 0.48002 9.71867 0.419979 9.84902C0.359935 9.97938 0.326391 10.1204
      0.321296 10.2638C0.3162 10.4072 0.339654 10.5502 0.390295 10.6845C0.440937 10.8188 0.517753 10.9417 0.616289 11.046C0.714822
      11.1504 0.833108 11.2341 0.964275 11.2924C1.09544 11.3506 1.23688 11.3822 1.38036 11.3853C1.52384 11.3885 1.66652 11.363 1.8001
      11.3106C1.93367 11.2581 2.05549 11.1796 2.15848 11.0796L11.1664 2.57329L20.1721 11.0796Z" fill={fill ? fill : '#212121'} />
    </svg>
  );
}

ArrowUp.defaultProps = {
  fill: undefined,
};
ArrowUp.propTypes = {
  fill: PropTypes.string,
};
