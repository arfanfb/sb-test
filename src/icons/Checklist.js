import React from 'react';
import PropTypes from 'prop-types';

export default function Checklist(props) {
  const { fill } = props;
  return (
    <svg fill="none" height="18" width="18" xmlns="http://www.w3.org/2000/svg" {...props}>
      <path clipRule="evenodd" d={`M16.1688 2.55667C16.0698 2.4821 15.957 2.42783 15.837 2.39699C15.7169
        2.36616 15.592 2.35936 15.4693 2.377C15.3466 2.39463 15.2286 2.43635 15.1221 2.49975C15.0155 2.56315
        14.9226 2.64698 14.8486 2.74642L7.01098 13.2428L3.06357 9.27811C2.97627 9.19004 2.8724 9.12013
        2.75794 9.07242C2.64348 9.02472 2.5207 9.00015 2.3967 9.00015C2.27269 9.00015 2.14992 9.02472 2.03546
        9.07242C1.921 9.12013 1.81712 9.19004 1.72983 9.27811C1.55269 9.45603 1.45325 9.69687 1.45325
        9.94793C1.45325 10.199 1.55269 10.4398 1.72983 10.6178L6.4463 15.3551C6.53363 15.443 6.63751 15.5129
        6.75196 15.5605C6.86641 15.6081 6.98916 15.6326 7.11313 15.6326C7.13547 15.6326 7.15781 15.6319 7.18015
        15.6305C7.31511 15.6207 7.44639 15.582 7.56501 15.5169C7.68364 15.4518 7.78682 15.3618 7.86752
        15.2532L16.3578 3.88387C16.5081 3.68259 16.5726 3.42999 16.5372 3.18129C16.5018 2.93258 16.3693
        2.70801 16.1688 2.55667Z`} fill={fill ? fill : '#DE1B1B'}  fillRule="evenodd"/>
    </svg>
  );
}

Checklist.defaultProps = {
  fill: undefined,
};
Checklist.propTypes = {
  fill: PropTypes.string,
};
