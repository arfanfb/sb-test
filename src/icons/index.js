import ArrowUp from './ArrowUp';
import Checklist from './Checklist';
import Logo from './Logo';

export {
  ArrowUp,
  Checklist,
  Logo,
};
