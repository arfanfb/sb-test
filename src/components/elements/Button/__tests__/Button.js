import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ArrowUp } from '../../../../icons';
import Button from '../Button';

jest.mock('react', () => ({
  ...require.requireActual('react'),
}));

describe('src/components/button', () => {
  it('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Button />);
    expect(tree).toMatchSnapshot();
  });

  it('expect width 480px and text Hello', () => {
    const result = Button({
      ...Button.defaultProps,
      width: 480,
      children: 'Hello',
    });
    const { children, style } = result.props;
    expect(children[2]).toBe('Hello');
    expect(style.width).toBe(480);
  });

  it('expect with link href and object', () => {
    const result = Button({
      ...Button.defaultProps,
      to: '/home',
      children: 'Link',
    });

    const { children } = result.props;
    expect(children.props.children[2]).toBe(undefined);

    const result2 = Button({
      ...Button.defaultProps,
      children: <ArrowUp />,
    });
    expect(typeof result2.props.children[2].type).toEqual('function');
  });
});
