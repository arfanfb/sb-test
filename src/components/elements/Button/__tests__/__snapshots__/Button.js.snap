// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/button render 1`] = `
<button
  className=""
  disabled={false}
  onClick={null}
  style={
    Object {
      "height": 48,
      "width": 182,
    }
  }
  type="button"
>
   
</button>
`;
